src/System.Private.ServiceModel/tools/test/SelfHostWcfService/TestResources/WebSocketResources.cs

```diff
@@ -118,4 +119,88 @@ protected override Binding GetBinding()
             return binding;
         }
     }
+
+    internal class WebSocketHttpsDuplexTextStreamedResource : EndpointResource<WSDuplexService, IWSDuplexService>
+    {
+        protected override string Protocol { get { return BaseAddressResource.Https; } }
+
+        protected override int GetPort(ResourceRequestContext context)
+        {
+            return context.BridgeConfiguration.BridgeHttpsPort;
```

Comment:
Since it was an https scenario I wasn't sure which one to use, I didn't want to add a new one. So either of those two make sense to me.